modal_proto/api.proto

```diff
@@ -408,6 +408,8 @@ message Function {
   string runtime = 30;
 
   string stub_name = 31;
+
+  bool is_builder_function = 32;
```

Comment:
maybe this should be some sort of function type? but we can revisit this once we have more types

Reply:
Yeah I was thinking of making it another enum value of the existing function type, but it felt semi-orthogonal to the existing values of function and generator. Agree we can revisit this later